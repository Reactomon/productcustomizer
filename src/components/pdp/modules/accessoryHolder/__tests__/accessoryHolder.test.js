import React from 'react';
import { shallow } from 'enzyme';
import { Link } from '@material-ui/core';
import AccessoryHolder from '../accessoryHolder';


describe('AccessoryHolder', () => {
    const testFunction = jest.fn();
    const component = shallow(
        <AccessoryHolder
            title="Primary Heading"
            accessorySuggestion='Suggestions Data'
            data={[
                {
                    selected: true,
                    additionalAmountRequired: 200,
                    sign: '+'
                }
            ]}
            setDefault={testFunction}
            defaultConfig={[]}
            handleShowModal={testFunction}
            descriptionIndex={0}
            handleChange={testFunction}
            modalData={{}}
        />);

    it('should render Accessory Holder', () => {
        expect(component).toBeDefined();
    });

    it('should render Accessory Holder', () => {
        const comp = shallow(
            <AccessoryHolder handleShowModal={null} />
        );
        console.log(comp.find(Link));
        expect(comp).toBeDefined();
    });
});

