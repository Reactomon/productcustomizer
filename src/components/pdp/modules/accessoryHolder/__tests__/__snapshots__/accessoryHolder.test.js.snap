// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessoryHolder should render Accessory Holder 1`] = `
<Fragment>
  <styled.h2>
    Primary Heading
  </styled.h2>
  <Styled(WithStyles(ForwardRef(Link)))
    display="block"
    onClick={[Function]}
  >
    Suggestions Data
  </Styled(WithStyles(ForwardRef(Link)))>
</Fragment>
`;

exports[`AccessoryHolder should render Accessory Holder 2`] = `
<Fragment>
  <styled.h2>
    Primary Heading
  </styled.h2>
  <Styled(WithStyles(ForwardRef(Link)))
    display="block"
    onClick={[Function]}
  >
    Suggestions Data
  </Styled(WithStyles(ForwardRef(Link)))>
</Fragment>
`;

exports[`Addition should render Accessory Holder 1`] = `
<Fragment>
  <styled.h2>
    Primary Heading
  </styled.h2>
  <Styled(WithStyles(ForwardRef(Link)))
    display="block"
    onClick={[Function]}
  >
    Suggestions Data
  </Styled(WithStyles(ForwardRef(Link)))>
</Fragment>
`;

exports[`should render Accessory Holder 1`] = `
<Fragment>
  <styled.h2>
    Primary Heading
  </styled.h2>
  <Styled(WithStyles(ForwardRef(Link)))
    display="block"
    onClick={[Function]}
  >
    Suggestions Data
  </Styled(WithStyles(ForwardRef(Link)))>
</Fragment>
`;
